public/pages/SampleData/containers/SampleData/SampleData.tsx

```diff
@@ -0,0 +1,252 @@
+  const [isLoadingEcommerceData, setIsLoadingEcommerceData] = useState<boolean>(
+    false
+  );
+  const [isLoadingHostHealthData, setIsLoadingHostHealthData] = useState<
+    boolean
+  >(false);
+
+  const getAllDetectors = async () => {
+    await dispatch(getDetectorList(GET_ALL_DETECTORS_QUERY_PARAMS)).catch(
+      (error: any) => {
+        console.error('Error getting all detectors: ', error);
+      }
+    );
+  };
+
+  const getAllIndices = async () => {
+    await dispatch(getIndices('')).catch((error: any) => {
+      console.error('Error getting all indices: ', error);
+    });
+  };
+
+  // Set breadcrumbs on page initialization
+  useEffect(() => {
+    chrome.breadcrumbs.set([
+      BREADCRUMBS.ANOMALY_DETECTOR,
+      BREADCRUMBS.SAMPLE_DETECTORS,
+    ]);
+  }, []);
+
+  // Getting all initial detectors
+  useEffect(() => {
+    getAllDetectors();
+    getAllIndices();
+  }, []);
+
+  // Create and populate sample index, create and start sample detector
+  const handleLoadData = async (
+    sampleType: SAMPLE_TYPE,
+    indexConfig: any,
+    detectorConfig: any,
+    setLoadingState: (isLoading: boolean) => void
+  ) => {
+    setLoadingState(true);
+    let errorDuringAction = false;
+    let errorMessage = '';
+
+    // Create the index (if it doesn't exist yet)
+    if (!containsSampleIndex(visibleIndices, sampleType)) {
+      await dispatch(createIndex(indexConfig)).catch((error: any) => {
+        errorDuringAction = true;
+        errorMessage = 'Error creating sample index.';
+        console.error('Error creating sample index: ', error);
+      });
+    }
+
+    // Get the sample data from the server and bulk insert
+    if (!errorDuringAction) {
+      await dispatch(createSampleData(sampleType)).catch((error: any) => {
+        errorDuringAction = true;
+        errorMessage = error;
+        console.error('Error creating sample detector: ', error);
+      });
+    }
+
+    // Create the detector
+    if (!errorDuringAction) {
+      await dispatch(createDetector(detectorConfig))
+        .then(function (response: any) {
+          const detectorId = response.data.response.id;
+          // Start the detector
+          dispatch(startDetector(detectorId)).catch((error: any) => {
+            errorDuringAction = true;
+            errorMessage = error.data.message;
+            console.error('Error starting sample detector: ', error);
+          });
+        })
+        .catch((error: any) => {
+          errorDuringAction = true;
+          errorMessage = error;
+          console.error('Error creating sample detector: ', error);
+        });
+    }
+
+    getAllDetectors();
+    getAllIndices();
+    setLoadingState(false);
+    if (!errorDuringAction) {
+      toastNotifications.addSuccess('Successfully loaded sample detector');
+    } else {
+      toastNotifications.addDanger(
+        `Unable to load all sample data. ${errorMessage}`
```

Comment:
Hard to say, since error could be from many different sources. In most cases, just trying to perform the actions again is the recommended action. Think we should change the toast to say something like `Unable to load all sample data, please try again. ${errorMessage}`

Reply:
Yes, this is the best we can do.  Since this may happen at any step, would it cause issues if we retry?  For example, if sample indices are already loaded, retry loading sample index would cause another error.

Reply:
I handle the case where if sample indices are already loaded, we skip the step of trying to create again, and go straight to creating a detector on it (could be a common case where index and detector initially created, sample detector deleted, want to recreate the detector again).

If the error is creating the index or creating the detector, this would be something unexpected where hopefully a retry would work. At least propagating the custom error message should help the user out a bit, since the error can be from several sources.